extensions/amp-a4a/0.1/test/test-amp-a4a.js

```diff
@@ -403,7 +403,9 @@ describe('amp-a4a', () => {
         const doc = fixture.doc;
         const a4aElement = doc.createElement('amp-a4a');
         const a4a = new AmpA4A(a4aElement);
-        return a4a.maybeRenderAmpAd_(null).then(rendered => {
+        a4a.adUrl_ = 'http://foo.com';
+        a4a.maybeRenderAmpAd_ = function() { return Promise.resolve(false); }
```

Comment:
I went this route because the current documentation indicates ArrayBuffer
input cannot be null which given the code the calls it first checks that it
is not null makes sense to me:

return creative && this.maybeRenderAmpAd_(creative);

/**
   * Render a validated AMP creative directly in the parent page.
   * @param {!ArrayBuffer} bytes The creative, as raw bytes.
   * @return {Promise<boolean>} Whether the creative was successfully
   *     rendered.
   * @private
   */
  maybeRenderAmpAd_(bytes) { ... }

On Fri, Jul 29, 2016 at 11:30 AM, Terran Lane <notifications@github.com>
wrote:

> In extensions/amp-a4a/0.1/test/test-amp-a4a.js
> <https://github.com/ampproject/amphtml/pull/4215#discussion_r72811353>:
>
> > @@ -403,7 +403,9 @@ describe('amp-a4a', () => {
> >          const doc = fixture.doc;
> >          const a4aElement = doc.createElement('amp-a4a');
> >          const a4a = new AmpA4A(a4aElement);
> > -        return a4a.maybeRenderAmpAd_(null).then(rendered => {
> > +        a4a.adUrl_ = 'http://foo.com';
> > +        a4a.maybeRenderAmpAd_ = function() { return Promise.resolve(false); }
>
> I don't think this is the right fix. maybeRenderAmpAd_() *should* be able
> to handle a null input. That should cause utf8FromArrayBuffer() to throw
> an error. It looks like the missing bit is that utf8FromArrayBuffer() is
> missing a .catch() clause. (Alternatively, maybeRenderAmpAd_() could just
> check for null on input and return Promise.resolve(false) immediately.)
>
> I could make that change if you like, or you could on this PR. Happy
> either way.
>
> What puzzles me is that this test *passed* when I submitted the previous
> PR. (Otherwise it wouldn't have gone in.) Once again, I'm suspicious of the
> unit test framework's reliability. :-(
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ampproject/amphtml/pull/4215/files/b61b2e8d312d2b3918f6723a42c14d28ac6728ee..ede9c7390fa511bd7f1fa11e994aa43e157ba889#r72811353>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABr0LyXRo0Lfe0kYrRH8Jpizdvccvx8Gks5qahyEgaJpZM4JVYV9>
> .
>